utils/envvars/parser.go

```diff
@@ -141,6 +148,44 @@ func parseStringDuration(value string) (time.Duration, error) {
 	return d, nil
 }
 
+// GetSAASHost returns SAAS host env variable value if it's valid.
+// Otherwise returns defaultSAASHost.
+func GetSAASHost(env string) (string, error) {
+	if v, ok := os.LookupEnv(envSAASHost); ok {
+		h, p, err := net.SplitHostPort(v)
+		if err != nil {
+			return v, err
+		}
+		if h != "" && p != "" {
+			logrus.Warnf("SAAS host changed to %q.", v)
+
+			return v, nil
+		}
+
+		return v, fmt.Errorf("environment variable %q has invalid format %q. Expected {host}:{port}", env, v)
```

Comment:
Just stylistic simplification, both conditions check for the error case

```suggestion
		h, p, err := net.SplitHostPort(v)
		if err != nil {
			return v, err
		}
		if h == "" || p == "" {
			return v, fmt.Errorf("environment variable %q has invalid format %q. Expected {host}:{port}", env, v)
		}
		
		logrus.Warnf("SAAS host changed to %q.", v)
		return v, nil
```

Reply:
this piece of code was updated by Aleksi, thank you